### PR TITLE
refactor: splitup table_functions.h

### PR DESCRIPTION
**Pull request by Riolku**

### CODE REVIEW

**Review by andyfengHKU — APPROVED** [review]

File: src/include/function/table_functions/show_tables.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "function/table_functions.h"
+#include "function/table_functions/bind_data.h"
+#include "function/table_functions/bind_input.h"
+
+namespace kuzu::function {
```
Comment (andyfengHKU) [inline src/include/function/table_functions/show_tables.h:7]: Though our convention is to use

namespace kuzu
namespace function

Let's stick to this. Or we have to change every place

Comment (Riolku) [reply src/include/function/table_functions/show_tables.h:7]: I strongly prefer this newer syntax. Shall we poll? I can change it everywhere in a subsequent PR.

Comment (andyfengHKU) [reply src/include/function/table_functions/show_tables.h:7]: sure let's bring to the standup

Comment (Riolku) [reply src/include/function/table_functions/show_tables.h]: Reverted namespace changes and will merge when tests pass.

**Pull request merged**